eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -103,16 +109,36 @@ jobs:
         parameters:
           Artifacts: ${{parameters.Artifacts}}
           AdditionalModules: ${{parameters.AdditionalModules}}
+      
+      - task: UsePythonVersion@0
+        displayName: 'Use Python 3.6'
+        inputs:
+          versionSpec: '3.6'
+        condition: and(succeeded(), eq(variables['TestFromSource'],'true'))
+
+      - task: PythonScript@0
+        displayName: 'Set versions for source build'
+        inputs:
+          scriptPath: 'eng/versioning/set_versions.py'
+          arguments: '--build-type client --pst'
+        condition: and(succeeded(), eq(variables['TestFromSource'],'true'))
+
+      - task: PythonScript@0
+        displayName: 'Update versions for source build'
+        inputs:
+          scriptPath: 'eng/versioning/update_versions.py'
+          arguments: '--update-type library --build-type client --sr'
+        condition: and(succeeded(), eq(variables['ShouldRunSourceTests'],'true'))
 
       - task: Maven@3
-        displayName: 'Build and Install, JDK Version: $(JavaBuildVersion)'
+        displayName: 'Build and Install, JDK Version: $(JavaTestVersion)'
         inputs:
           mavenPomFile: pom.xml
           goals: 'install'
-          options: '$(DefaultOptions) -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -pl $(ProjectList) -am'
+          options: '$(DefaultOptions) -T 1C -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -pl $(ProjectList) -am'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
-          jdkVersionOption: $(JavaBuildVersion)
+          jdkVersionOption: $(JavaTestVersion)
```

Comment:
Was this broken before?

Reply:
We were always building with Java 11 which resulted in the Java 8 runs needing to compile twice as the compiler version changed. Unfortunately, I'm not explicitly certain on why this was done.